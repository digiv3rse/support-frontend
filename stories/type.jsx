// @flow

import React from 'react';

import { storiesOf } from '@storybook/react';

import Text, { Title, Callout, LargeParagraph, SansParagraph } from 'components/text/text';
import UnorderedList from 'components/list/unorderedList';
import OrderedList from 'components/list/orderedList';

import { withCenterAlignment } from '../.storybook/decorators/withCenterAlignment';

const stories = storiesOf('Type', module)
  .addDecorator(withCenterAlignment);

stories.add('Text', () => (
  <div style={{ maxWidth: '30em' }}>
    <Text title="If you have to show text, wrap it in <Text>">
      <LargeParagraph>Easy, right?</LargeParagraph>
      <p>
        For your convenience you
        can add a heading right at the top but you can also use
        the <code>Title</code> export. up to you!
      </p>
      <SansParagraph>
        <code>Text</code> will format some stuff like links
        for you and you can nest it inside a wrapper.
      </SansParagraph>
      <Callout>It also exports just about everything you can put in it</Callout>
      <Title size={3}>Even more titles!</Title>
    </Text>
  </div>
));

stories.add('Lists', () => (
  <Text title="Lists">
    <UnorderedList items={[
      'This is an unordered list',
      'It looks like this',
    ]}
    />
    <OrderedList items={[
      'This one is ordered',
      'and it looks like this',
    ]}
    />
  </Text>
));
