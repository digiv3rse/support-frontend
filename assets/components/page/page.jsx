// @flow

// ----- Imports ----- //

import React, { type Node } from 'react';

import { classNameWithModifiers } from 'helpers/utilities';

import TimeTravelBanner from 'components/headerBanners/timeTravelBanner';


// ----- Types ----- //

type PropTypes = {|
  id: ?string,
  header: Node,
  footer: Node,
  children: Node,
  classModifiers: Array<?string>
|};


// ----- Component ----- //

export default function Page(props: PropTypes) {

  return (
    <div id={props.id} className={classNameWithModifiers('gu-content', props.classModifiers)}>
      <TimeTravelBanner />
      {props.header}
      <main role="main" className="gu-content__main">
        {props.children}
      </main>
      {props.footer}
    </div>
  );

}


// ----- Default Props ----- //

Page.defaultProps = {
  id: null,
  classModifiers: [],
};
