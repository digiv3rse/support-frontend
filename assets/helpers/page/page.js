// @flow

// ----- Imports ----- //

import { createStore, combineReducers } from 'redux';
import 'ophan';

import * as abTest from 'helpers/abTests/abtest';
import * as logger from 'helpers/logger';
import * as googleTagManager from 'helpers/tracking/googleTagManager';

import { getCampaign, getAcquisition } from 'helpers/tracking/acquisitions';
import { detect as detectCountryGroup } from 'helpers/internationalisation/countryGroup';
import { detect as detectCountry } from 'helpers/internationalisation/country';
import { detect as detectCurrency } from 'helpers/internationalisation/currency';

import type { Campaign, ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
import type { IsoCountry } from 'helpers/internationalisation/country';
import type { Currency } from 'helpers/internationalisation/currency';
import type { Participations } from 'helpers/abTests/abtest';

import { getAllQueryParamsWithExclusions } from 'helpers/url';
import type { Action } from './pageActions';


// ----- Types ----- //

export type CommonState = {
  campaign: ?Campaign,
  referrerAcquisitionData: ReferrerAcquisitionData,
  currency: Currency,
  otherQueryParams: Array<[string, string]>,
  countryGroup: CountryGroupId,
  country: IsoCountry,
  abParticipations: Participations,
};

export type PreloadedState = {
  campaign?: $PropertyType<CommonState, 'campaign'>,
  referrerAcquisitionData?: $PropertyType<CommonState, 'referrerAcquisitionData'>,
  country?: $PropertyType<CommonState, 'country'>,
  abParticipations?: $PropertyType<CommonState, 'abParticipations'>,
};


// ----- Functions ----- //

function doNotTrack(): boolean {
  // $FlowFixMe
  const doNotTrackFlag = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;

  return doNotTrackFlag === '1' || doNotTrackFlag === 'yes';
}

// Sets up GA and logging.
function analyticsInitialisation(participations: Participations): void {
  if (!(doNotTrack())) {
    googleTagManager.init(participations);
  }
  // Logging.
  logger.init();
}

// Creates the initial state for the common reducer.
function buildInitialState(
  abParticipations: Participations,
  preloadedState: ?PreloadedState = {},
  countryGroup: CountryGroupId,
  country: IsoCountry,
  currency: Currency,
): CommonState {
  const acquisition = getAcquisition(abParticipations);
  const excludedParameters = ['REFPVID', 'INTCMP', 'acquisitionData'];
  const otherQueryParams = getAllQueryParamsWithExclusions(excludedParameters);

  return Object.assign({}, {
    campaign: acquisition ? getCampaign(acquisition) : null,
    referrerAcquisitionData: acquisition,
    otherQueryParams,
    countryGroup,
    country,
    abParticipations,
    currency,
  }, preloadedState);

}

// Sets up the common reducer with its initial state.
function createCommonReducer(initialState: CommonState): (CommonState, Action) => CommonState {

  function commonReducer(
    state: CommonState = initialState,
    action: Action,
  ): CommonState {
    switch (action.type) {
      case 'SET_COUNTRY':
        return Object.assign({}, state, { country: action.country });
      case 'SET_COUNTRY_GROUP':
        return {
          ...state,
          countryGroup: action.countryGroup,
          currency: detectCurrency(action.countryGroup),
          country: detectCountry(action.countryGroup),
        };
      default:
        return state;
    }
  }

  return commonReducer;
}

// For pages that don't need Redux.
function statelessInit() {
  const country: IsoCountry = detectCountry();
  const participations: Participations = abTest.init(country);
  analyticsInitialisation(participations);

}

// Initialises the page.
function init(
  pageReducer: Object,
  preloadedState: ?PreloadedState = null,
  middleware: ?Function,
) {

  const countryGroup: CountryGroupId = detectCountryGroup();
  const country: IsoCountry = detectCountry();
  const currency: Currency = detectCurrency(countryGroup);
  const participations: Participations = abTest.init(country);
  analyticsInitialisation(participations);

  const initialState: CommonState = buildInitialState(
    participations,
    preloadedState,
    countryGroup,
    country,
    currency,
  );
  const commonReducer = createCommonReducer(initialState);

  return createStore(
    combineReducers({ page: pageReducer, common: commonReducer }),
    undefined,
    middleware,
  );
}


// ----- Exports ----- //

export {
  createCommonReducer,
  init,
  statelessInit,
};
