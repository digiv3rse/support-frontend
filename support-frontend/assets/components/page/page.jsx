// @flow

// ----- Imports ----- //

import React, { type Node } from 'react';
import { classNameWithModifiers } from 'helpers/utilities';
import TimeTravelBanner from 'components/headerBanners/timeTravelBanner';
import { connect } from 'react-redux';
import type { State, ThankYouPageStage } from 'pages/new-contributions-landing/contributionsLandingReducer';


// ----- Types ----- //

type PropTypes = {|
  id: ?string,
  header: Node,
  footer: Node,
  children: Node,
  classModifiers: Array<?string>,
  backgroundImageSrc: ?string,
  thankYouPageStage: ThankYouPageStage,
|};

const mapStateToProps = (state: State) => ({
  thankYouPageStage: state.page.form.thankYouPageStage,
});


// ----- Component ----- //

function Page(props: PropTypes) {
  // Set password screen shouldn't inherit new page-level classnames, unlike other pages in thank you flow
  const classMods = props.thankYouPageStage === 'thankYouSetPassword' ? [] : props.classModifiers;

  const backgroundImage = props.backgroundImageSrc ? (
    <div className="background-image-container">
      <img className="background-image" alt="landing page background illustration" src={props.backgroundImageSrc} />
    </div>
  ) : null;

  return (
    <div id={props.id} className={classNameWithModifiers('gu-content', classMods)}>
      <TimeTravelBanner />
      {props.header}
      <main role="main" className="gu-content__main">
        {backgroundImage}
        {props.children}
      </main>
      {props.footer}
    </div>
  );

}


// ----- Default Props ----- //

Page.defaultProps = {
  id: null,
  classModifiers: [],
  backgroundImageSrc: null,
};

export default connect(mapStateToProps)(Page);
