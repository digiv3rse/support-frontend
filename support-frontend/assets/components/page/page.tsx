// ----- Imports ----- //
// @ts-expect-error - required for hooks
import type { Node } from 'react';
import React, { useEffect } from 'react';
import CsrBanner from 'components/csr/csrBanner';
import { classNameWithModifiers } from 'helpers/utilities/utilities';
// ----- Types ----- //
type PropTypes = {
	id: string | null | undefined;
	header: Node;
	footer: Node | null;
	children: Node;
	classModifiers: Array<string | null | undefined>;
	backgroundImageSrc: string | null | undefined;
}; // ----- Component ----- //

export default function Page(props: PropTypes) {
	const backgroundImage = props.backgroundImageSrc ? (
		<div className="background-image-container">
			<img
				className="background-image"
				alt="landing page background illustration"
				src={props.backgroundImageSrc}
			/>
		</div>
	) : null;
	useEffect(() => {
		requestAnimationFrame(() => {
			if (window.location.hash) {
				const hashElement = document.getElementById(
					window.location.hash.substr(1),
				);

				if (hashElement) {
					hashElement.scrollIntoView();
				}
			}
		});
	}, []);
	return (
		<div
			id={props.id}
			className={classNameWithModifiers('gu-content', props.classModifiers)}
		>
			<CsrBanner />
			{props.header}
			<main role="main" className="gu-content__main">
				{backgroundImage}
				{props.children}
			</main>
			{props.footer}
		</div>
	);
} // ----- Default Props ----- //

Page.defaultProps = {
	id: null,
	footer: null,
	classModifiers: [],
	backgroundImageSrc: null,
};
