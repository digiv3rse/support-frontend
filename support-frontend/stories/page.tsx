import { radios, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import CentredContainer from 'components/containers/centredContainer';
import GridImage from 'components/gridImage/gridImage';
import Block from 'components/page/block';
import Hero from 'components/page/hero';
import HeroRoundel from 'components/page/heroRoundel';
import PageTitle from 'components/page/pageTitle';

const stories = storiesOf('Page composition components', module).addDecorator(
	withKnobs,
);
const DemoHero = (
	<Hero
		image={
			<GridImage
				gridId="weeklyCampaignHeroImg"
				srcSizes={[500]}
				sizes="(max-width: 740px) 500px, 500px"
				imgType="png"
				altText="A collection of Guardian Weekly magazines"
			/>
		}
		roundelText={
			<div>
				<div>First</div>
				<div>issue</div>
				<div>free</div>
			</div>
		}
	>
		<section
			style={{
				padding: '0 1rem',
			}}
		>
			<h1
				style={{
					fontSize: '42px',
					fontFamily: 'GH Guardian Headline',
				}}
			>
				Hero Heading
			</h1>
			<p
				style={{
					paddingTop: '1em',
				}}
			>
				Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque id
				justo at est elementum egestas rhoncus eu nulla. Proin pellentesque
				massa at metus condimentum, a aliquam erat condimentum. Vivamus quis
				rutrum nulla. Curabitur ut ullamcorper magna, eu ornare nunc.
			</p>
		</section>
	</Hero>
);
stories.add('Hero', () => (
	<div
		style={{
			width: '100%',
			paddingTop: '100px',
		}}
	>
		<CentredContainer>{DemoHero}</CentredContainer>
	</div>
));
stories.add('HeroRoundel', () => (
	<div
		style={{
			paddingTop: '100px',
		}}
	>
		<CentredContainer>
			<HeroRoundel theme="digital">
				<div>
					<div
						style={{
							fontSize: '42px',
						}}
					>
						14 day
					</div>
					<div>free trial</div>
				</div>
			</HeroRoundel>
		</CentredContainer>
	</div>
));
stories.add('Header', () => {
	const theme = radios(
		'Theme',
		{
			showcase: 'showcase',
			digital: 'digital',
			paper: 'paper',
			weekly: 'weekly',
		},
		'digital',
	);
	return (
		<PageTitle title="Page Title" theme={theme}>
			<CentredContainer>{DemoHero}</CentredContainer>
		</PageTitle>
	);
});
stories.add('Block', () => (
	<CentredContainer>
		<Block>
			<section
				style={{
					border: '1px solid black',
					padding: '1rem',
				}}
			>
				<p>This is a container for text or other content</p>
			</section>
		</Block>
	</CentredContainer>
));
