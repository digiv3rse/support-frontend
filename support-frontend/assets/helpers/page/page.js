// @flow

// ----- Imports ----- //

import * as ophan from 'ophan';
import type { Store } from 'redux';
import {
  applyMiddleware,
  combineReducers,
  compose,
  createStore,
  type Reducer,
} from 'redux';
import thunkMiddleware from 'redux-thunk';

import type { Participations } from 'helpers/abTests/abtest';
import * as abTest from 'helpers/abTests/abtest';
import { renderError } from 'helpers/render';
import type { Settings } from 'helpers/settings';
import * as logger from 'helpers/logger';
import * as googleTagManager from 'helpers/tracking/googleTagManager';
import {
  detect as detectCountry,
  type IsoCountry,
} from 'helpers/internationalisation/country';
import {
  detect as detectCurrency,
  type IsoCurrency,
} from 'helpers/internationalisation/currency';
import { getAllQueryParamsWithExclusions } from 'helpers/url';
import type { CommonState } from 'helpers/page/commonReducer';
import { createCommonReducer } from 'helpers/page/commonReducer';
import {
  getCampaign,
  getReferrerAcquisitionData,
} from 'helpers/tracking/acquisitions';
import {
  type CountryGroupId,
  detect as detectCountryGroup,
} from 'helpers/internationalisation/countryGroup';
import { trackAbTests, setReferrerDataInLocalStorage } from 'helpers/tracking/ophan';
import { getSettings } from 'helpers/globals';
import { getGlobal } from 'helpers/globals';
import { isPostDeployUser } from 'helpers/user/user';
import { getAmounts } from 'helpers/abTests/helpers';
import type { ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
import {localCurrencyCountries} from "../internationalisation/localCurrencyCountry";
import type {LocalCurrencyCountry} from "../internationalisation/localCurrencyCountry";

if (process.env.NODE_ENV === 'DEV') {
  // $FlowIgnore
  import('preact/debug');
}

// ----- Types ----- //

export type ReduxState<PageState> = {|
  common: CommonState,
  page: PageState,
|};


// ----- Functions ----- //

// Sets up GA and logging.
function analyticsInitialisation(participations: Participations, acquisitionData: ReferrerAcquisitionData): void {
  setReferrerDataInLocalStorage(acquisitionData);
  googleTagManager.init(participations);
  ophan.init();
  trackAbTests(participations);
  // Logging.
  logger.init();
}

// Creates the initial state for the common reducer.
function buildInitialState(
  abParticipations: Participations,
  countryGroupId: CountryGroupId,
  countryId: IsoCountry,
  currencyId: IsoCurrency,
  settings: Settings,
  acquisitionData: ReferrerAcquisitionData,
  localCurrencyCountry: LocalCurrencyCountry | null,
  useLocalCurrency: boolean,
): CommonState {
  const excludedParameters = ['REFPVID', 'INTCMP', 'acquisitionData'];
  const otherQueryParams = getAllQueryParamsWithExclusions(excludedParameters);
  const internationalisation = {
    countryGroupId,
    countryId,
    currencyId,
    localCurrencyCountry,
    useLocalCurrency,
  };

  const amounts = getAmounts(settings, abParticipations, countryGroupId);

  return {
    campaign: acquisitionData ? getCampaign(acquisitionData) : null,
    referrerAcquisitionData: acquisitionData,
    otherQueryParams,
    internationalisation,
    abParticipations,
    settings,
    amounts,
  };

}

// For pages that don't need Redux.
function statelessInit() {
  const country: IsoCountry = detectCountry();
  const countryGroupId: CountryGroupId = detectCountryGroup();
  const participations: Participations = abTest.init(country, countryGroupId, window.guardian.settings);
  const acquisitionData = getReferrerAcquisitionData();
  analyticsInitialisation(participations, acquisitionData);
}

// Enables redux devtools extension and optional redux-thunk.
/* eslint-disable no-underscore-dangle */
function storeEnhancer(thunk: boolean) {

  if (thunk) {
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    return composeEnhancers(applyMiddleware(thunkMiddleware));
  }

  return window.__REDUX_DEVTOOLS_EXTENSION__ ?
    window.__REDUX_DEVTOOLS_EXTENSION__()
    : undefined;

}
/* eslint-enable no-underscore-dangle */

// Initialises the page.
function init<S, A>(
  pageReducer?: CommonState => Reducer<S, A> | null,
  thunk?: boolean = false,
): Store<*, *, *> {
  try {
    const countryId: IsoCountry = detectCountry();

    /**
     * Dynamically load @guardian/consent-management-platform
     * on condition we're not server side rendering (ssr) the page.
     * @guardian/consent-management-platform breaks ssr otherwise.
     */
    if (!getGlobal('ssr') && !isPostDeployUser()) {
      import('@guardian/consent-management-platform').then(({ cmp }) => {
        cmp.init({
          country: countryId,
        });
      });
    }

    const acquisitionData = getReferrerAcquisitionData();

    const settings = getSettings();
    const countryGroupId: CountryGroupId = detectCountryGroup();
    const currencyId: IsoCurrency = detectCurrency(countryGroupId);
    const participations: Participations = abTest.init(countryId, countryGroupId, settings);
    analyticsInitialisation(participations, acquisitionData);

    const initialState: CommonState = buildInitialState(
      participations,
      countryGroupId,
      countryId,
      currencyId,
      settings,
      acquisitionData,
      localCurrencyCountries[window.guardian.geoip.countryCode],
      false,
    );
    const commonReducer = createCommonReducer(initialState);

    const store = createStore(
      combineReducers({ page: pageReducer ? pageReducer(initialState) : null, common: commonReducer }),
      storeEnhancer(thunk),
    );

    return store;
  } catch (err) {
    renderError(err, null);
    throw (err);
  }
}


// ----- Exports ----- //

export {
  init,
  statelessInit,
};
