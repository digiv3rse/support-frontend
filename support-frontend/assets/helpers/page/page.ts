// ----- Imports ----- //
import type { Store } from "redux";
import type { Reducer } from "redux";
import { applyMiddleware, combineReducers, compose, createStore } from "redux";
import thunkMiddleware from "redux-thunk";
import type { Participations } from "helpers/abTests/abtest";
import * as abTest from "helpers/abTests/abtest";
import { renderError } from "helpers/rendering/render";
import type { Settings } from "helpers/globalsAndSwitches/settings";
import type { IsoCountry } from "helpers/internationalisation/country";
import { detect as detectCountry } from "helpers/internationalisation/country";
import type { IsoCurrency } from "helpers/internationalisation/currency";
import { detect as detectCurrency } from "helpers/internationalisation/currency";
import { getAllQueryParamsWithExclusions, getQueryParameter } from "helpers/urls/url";
import type { CommonState, Internationalisation } from "helpers/page/commonReducer";
import { createCommonReducer } from "helpers/page/commonReducer";
import { getCampaign, getReferrerAcquisitionData } from "helpers/tracking/acquisitions";
import type { CountryGroupId } from "helpers/internationalisation/countryGroup";
import { detect as detectCountryGroup } from "helpers/internationalisation/countryGroup";
import { getSettings } from "helpers/globalsAndSwitches/globals";
import { getAmounts } from "helpers/abTests/helpers";
import type { ReferrerAcquisitionData } from "helpers/tracking/acquisitions";
import { localCurrencyCountries } from "../internationalisation/localCurrencyCountry";
import type { LocalCurrencyCountry } from "../internationalisation/localCurrencyCountry";
import { analyticsInitialisation, consentInitialisation } from "helpers/page/analyticsAndConsent";
// ----- Types ----- //
export type ReduxState<PageState> = {
  common: CommonState;
  page: PageState;
};

// ----- Functions ----- //
function getLocalCurrencyCountry(countryId: IsoCountry, abParticipations: Participations): LocalCurrencyCountry | null | undefined {
  const queryParam = getQueryParameter('local-currency-country');

  if (queryParam) {
    return localCurrencyCountries[queryParam.toUpperCase()];
  }

  if (abParticipations.localCurrencyTestV2 === 'variant') {
    return localCurrencyCountries[countryId];
  }

  return null;
}

// Creates the initial state for the common reducer.
function buildInitialState(abParticipations: Participations, countryGroupId: CountryGroupId, countryId: IsoCountry, currencyId: IsoCurrency, settings: Settings, acquisitionData: ReferrerAcquisitionData): CommonState {
  const excludedParameters = ['REFPVID', 'INTCMP', 'acquisitionData'];
  const otherQueryParams = getAllQueryParamsWithExclusions(excludedParameters);
  const localCurrencyCountry = getLocalCurrencyCountry(countryId, abParticipations);
  const internationalisation: Internationalisation = {
    countryGroupId,
    countryId,
    currencyId,
    useLocalCurrency: false,
    defaultCurrency: currencyId
  };

  if (localCurrencyCountry) {
    internationalisation.localCurrencyCountry = localCurrencyCountry;
  }

  const amounts = getAmounts(settings, abParticipations, countryGroupId);
  return {
    campaign: acquisitionData ? getCampaign(acquisitionData) : null,
    referrerAcquisitionData: acquisitionData,
    otherQueryParams,
    internationalisation,
    abParticipations,
    settings,
    amounts,
    defaultAmounts: amounts
  };
}

// Enables redux devtools extension and optional redux-thunk.

/* eslint-disable no-underscore-dangle */
function storeEnhancer(thunk: boolean) {
  if (thunk) {
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    return composeEnhancers(applyMiddleware(thunkMiddleware));
  }

  return window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : undefined;
}

/* eslint-enable no-underscore-dangle */
// Initialises the page.
function initRedux<S, A>(pageReducer?: (arg0: CommonState) => Reducer<S, A> | null, thunk: boolean = false): Store<any, any, any> {
  try {
    const countryId: IsoCountry = detectCountry();
    const countryGroupId: CountryGroupId = detectCountryGroup();
    const currencyId: IsoCurrency = detectCurrency(countryGroupId);
    const settings = getSettings();
    const participations: Participations = abTest.init(countryId, countryGroupId, settings);
    const acquisitionData: ReferrerAcquisitionData = getReferrerAcquisitionData();
    const initialState: CommonState = buildInitialState(participations, countryGroupId, countryId, currencyId, settings, acquisitionData);
    const commonReducer = createCommonReducer(initialState);
    const store = createStore(combineReducers({
      page: pageReducer ? pageReducer(initialState) : null,
      common: commonReducer
    }), storeEnhancer(thunk));
    return store;
  } catch (err) {
    renderError(err, null);
    throw err;
  }
}

function setUpTrackingAndConsents() {
  const countryId: IsoCountry = detectCountry();
  const countryGroupId: CountryGroupId = detectCountryGroup();
  const settings = getSettings();
  const participations: Participations = abTest.init(countryId, countryGroupId, settings);
  const acquisitionData = getReferrerAcquisitionData();
  consentInitialisation(countryId);
  analyticsInitialisation(participations, acquisitionData);
}

// ----- Exports ----- //
export { initRedux, setUpTrackingAndConsents };