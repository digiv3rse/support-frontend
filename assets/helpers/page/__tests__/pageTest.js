// @flow

// ----- Imports ----- //

import type { IsoCountry } from 'helpers/internationalisation/country';

import { createCommonReducer } from '../page';


// ----- Tests ----- //

jest.mock('ophan', () => {});

describe('reducer tests', () => {

  beforeEach(() => {

    const initialState = {
      campaign: 'dummy_campaign',
      referrerAcquisitionData: {
        referrerPageviewId: null,
        campaignCode: null,
        componentId: null,
        componentType: null,
        source: null,
        abTests: [],
        queryParameters: [],
      },
      internationalisation: {
        countryId: 'GB',
        countryGroupId: 'GBPCountries',
        currencyId: 'GBP',
      },
      abParticipations: {},
      otherQueryParams: [],
      settings: {
        switches: {},
      },
      optimizeExperiments: {},
    };

    global.reducer = createCommonReducer(initialState);

  });

  it('should return the initial state', () => {
    expect(global.reducer(undefined, { type: 'SET_COUNTRY', country: 'GB' })).toMatchSnapshot();
  });

  it('should handle SET_COUNTRY to US', () => {

    const country: IsoCountry = 'US';
    const action = {
      type: 'SET_COUNTRY',
      country,
    };

    const newState = global.reducer(undefined, action);

    expect(newState && newState.internationalisation.countryId).toEqual(country);
    expect(newState && newState.referrerAcquisitionData.campaignCode).toMatchSnapshot();
    expect(newState && newState.campaign).toMatchSnapshot();
    expect(newState &&
      newState.referrerAcquisitionData &&
      newState.referrerAcquisitionData.campaignCode).toMatchSnapshot();
    expect(newState && newState.abParticipations).toMatchSnapshot();
  });

});
