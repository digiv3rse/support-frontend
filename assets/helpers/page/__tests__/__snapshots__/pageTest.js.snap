// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer tests should handle SET_COUNTRY to US 1`] = `null`;

exports[`reducer tests should handle SET_COUNTRY to US 2`] = `"dummy_campaign"`;

exports[`reducer tests should handle SET_COUNTRY to US 3`] = `null`;

exports[`reducer tests should handle SET_COUNTRY to US 4`] = `Object {}`;

exports[`reducer tests should return the initial state 1`] = `
Object {
  "abParticipations": Object {},
  "campaign": "dummy_campaign",
  "internationalisation": Object {
    "countryGroupId": "GBPCountries",
    "countryId": "GB",
    "currencyId": "GBP",
  },
  "optimizeExperiments": Object {},
  "otherQueryParams": Array [],
  "referrerAcquisitionData": Object {
    "abTests": Array [],
    "campaignCode": null,
    "componentId": null,
    "componentType": null,
    "queryParameters": Array [],
    "referrerPageviewId": null,
    "source": null,
  },
  "settings": Object {
    "switches": Object {},
  },
}
`;
